pkg/cloud/ratelimit_test.go

```diff
@@ -82,3 +82,143 @@ func TestMinimumRateLimiter(t *testing.T) {
 		t.Errorf("`called` = true, want false")
 	}
 }
+
+func TestTickerRateLimiter(t *testing.T) {
+	t.Parallel()
+
+	trl := NewTickerRateLimiter(100, time.Second)
+	start := time.Now()
+	for i := 0; i < 50; i++ {
+		err := trl.Accept(context.Background(), nil)
+		if err != nil {
+			t.Errorf("TickerRateLimiter.Accept = %v, want nil", err)
+		}
+	}
+	elapsed := time.Since(start)
+	if elapsed > time.Second {
+		t.Errorf("TickerRateLimiter.Accept took too long: %v, want <1s", elapsed)
+	}
+	if elapsed < 500*time.Millisecond {
+		t.Errorf("TickerRateLimiter.Accept took too short: %v, want >500ms", elapsed)
+	}
+
+	// Use context that has been cancelled and expect a context error returned.
+	ctxCancelled, cancelled := context.WithCancel(context.Background())
+	cancelled()
+	// Verify context is cancelled by now.
+	<-ctxCancelled.Done()
+	err := trl.Accept(ctxCancelled, nil)
+	if err != ctxCancelled.Err() {
+		t.Errorf("AcceptRateLimiter.Accept() = %v, want %v", err, ctxCancelled.Err())
+	}
+}
+
+func TestCompositeRateLimiter(t *testing.T) {
```

Comment:
I think this should be parameter test where you register rl with different RL key.
you need to combine all keys in map (project, service, operation) and check that correct rl was chosen.

Reply:
That's what I do in `TestCompositeRateLimiter_Table` below